server/lib/nl_page_config.py

```diff
@@ -107,61 +107,67 @@ def _single_place_multiple_var_timeline_block(svs, sv2name):
 
 def build_page_config(detection: Detection, data_spec: DataSpec,
                       context_history):
-  all_svs = data_spec.selected_svs + data_spec.expanded_svs
-  for _, svs in data_spec.extended_sv_map.items():
-    all_svs.extend(svs)
-  sv2name = get_sv_name(all_svs)
+
+  main_place_spec = data_spec.main_place_spec
+  contained_place_spec = data_spec.contained_place_spec
+
   # Init
   page_config = subject_page_pb2.SubjectPageConfig()
   # Set metadata
-  page_config.metadata.place_dcid.append(data_spec.main.place)
+  page_config.metadata.place_dcid.append(main_place_spec.place)
   page_config.metadata.contained_place_types[
-      data_spec.main.type] = data_spec.contained.contained_place_type
+      main_place_spec.type] = contained_place_spec.contained_place_type
 
   # Set category data
   category = page_config.categories.add()
   classifier = detection.classifications[0]
   classificationType = classifier.type
 
   # No stat vars found
-  # TODO: use context to find previous stat vars
-  if not data_spec.selected_svs:
+  primary_sv = data_spec.primary_sv
+  primary_sv_siblings = data_spec.primary_sv_siblings
+  if not primary_sv:
+    for context in context_history:
+      if context['debug']['primary_sv']:
+        primary_sv = context['debug']['primary_sv']
+        primary_sv_siblings = context['debug']['primary_sv_siblings']
+        break
+
+  if not primary_sv:
     return page_config
 
+  all_svs = [primary_sv] + primary_sv_siblings
+  sv2name = get_sv_name(all_svs)
+
   if classificationType in [
       ClassificationType.SIMPLE, ClassificationType.OTHER
   ]:
     # The primary stat var
-    primary_sv = data_spec.selected_svs[0]
     block, stat_var_spec_map = _single_place_single_var_timeline_block(
         primary_sv, sv2name)
     category.blocks.append(block)
     for sv_key, spec in stat_var_spec_map.items():
       category.stat_var_spec[sv_key].CopyFrom(spec)
 
     # The siblings for the primary stat var
-    usable_sibiling_svs = [
-        x for x in data_spec.extended_sv_map[primary_sv]
-        if x in data_spec.main.svs
-    ]
-    if usable_sibiling_svs:
```

Comment:
If I'm reading this right, is main_place_spec.svs not relied on?

Reply:
Right, in the bottom of server/lib/nl_data_spec.py, I noted the primary_sv does not depend on the place.

This is definitely not handled well now. For two queries with two places, they could have different svs with data, so the renderable sv list depend on the place. This is a super simplification now.

Reply:
I see. Thanks.